kernel/core/utils/stringtools.hpp

```diff
@@ -130,6 +131,241 @@ enum TimeDateFormat
+    }
+
+    /////////////////////////////////////////////////
+    /// \brief This function is called for every found seperator and adds to the tape
+    ///
+    /// \param sep char
+    /// \param idx int
+    /// \return void
+    ///
+    /////////////////////////////////////////////////
+    void addSeperator(char sep, int idx)
+    {
+        pushInbetween(idx-1);
+        m_tape.push_back(sep);
+        m_last_idx = idx;
+
+        // max 3 symbols at ones in tape
+        while(m_tape.length() > 3)
+            m_tape.erase(m_tape.begin());
+
+        m_curr_format |= checkNumFormat(m_tape);
+    }
+
+    /////////////////////////////////////////////////
+    /// \brief This function is called when last char of a number is reached during parsing
+    ///
+    /// \param idx int
+    /// \return int
+    ///
+    /////////////////////////////////////////////////
+    int endParseNumber(int idx)
+    {
+        // we only want to check for the last 2/1
+        m_tape = m_tape.back();
+
+        pushInbetween(idx); //since normally the function is called when seperator appears
+        m_curr_format |= checkNumFormat(m_tape);
+
+        m_last_idx = idx;
+        m_tape = "";
+        return m_curr_format;
+    }
+
+    /////////////////////////////////////////////////
+    /// \brief This function is called to make a vote with the given number Type
+    ///
+    /// \param numType int
+    /// \return void
+    ///
+    /////////////////////////////////////////////////
+    void vote(int numType)
+    {
+        if((numType & NUM_DECIMAL_EU && numType & NUM_K_US) ||
+                (numType & NUM_DECIMAL_US && numType & NUM_K_EU))
+        {
+            m_num_format_votes[FMT_INVALID]++;  // INVALID Combination
+        }
+        else
+        {
+            if(numType & NUM_DECIMAL_EU)
+                m_num_format_votes[FMT_DEC_EU]++;
+            else if(numType & NUM_DECIMAL_US)
+                m_num_format_votes[FMT_DEC_US]++;
+
+            if(numType & NUM_K_EU)
+                m_num_format_votes[FMT_K_EU]++;
+            else if(numType & NUM_K_US)
+                m_num_format_votes[FMT_K_US]++;
+            else if(numType & NUM_K_SPACE)
+                m_num_format_votes[FMT_K_SPACE]++;
+
+            if(numType & NUM_INVALID)
+                m_num_format_votes[FMT_INVALID]++;
+        }
+    }
+
+    /////////////////////////////////////////////////
+    /// \brief This function returns the current Format given by votes
+    ///
+    /// \return int
+    ///
+    /////////////////////////////////////////////////
+    int getFormat()
+    {
+        // As soon as there are more than 2 set, we will have at least one bad sample that would be parsed wrong
+        int inval_cnt = 0;
+        for(int i = 0; i < FMT_COUNT; i++)
+        {
+            if(m_num_format_votes[i] > 0)
+                inval_cnt++;
+        }
+        if(inval_cnt > 2 || m_num_format_votes[FMT_INVALID] > 0)
+            return NUM_INVALID;

```

Comment:
Looks a bit over-engineered. Why do we need to count for number of votes? Isn't that enough:
```suggestion
        if (m_num_format_votes[FMT_INVALID])
            return NUM_INVALID;
```

Reply:
If our objective is to obtain the format with the highest number of votes, then yes. 
However, if there are votes for both the US and the EU, it is certain that one of them will be parsed incorrectly. The idea behind the "> 2" check was to not convert anything when different formats emerge within the same column.

Reply:
I'm not sure, whether I understood that correctly, but how does adding up *all* votes help with the idea to filter out invalid combinations? And then: if only one format is invalid, then the result of the whole vote is invalid, so there's no reason in *counting* the votes (at this point).

Reply:
It is not adding up all votes, it is a counter of how many different votes we have.
Then the idea was that we should max have 2 different types of votes, decimal & thousands separator.
However, as this would consider two decimal or two thousand separators as valid, this code does not work as we want.

Just for clarification, if we encounter more than one number format in an entire column, we don't want to proceed with the conversion to `CONVTYPE_VALUE` at all, correct? In that case, we should return `NUM_INVALID` here.

Reply:
You're right, I was probably confused by the variable name. If those two number formats exclude each other and there's no possibility of a false detection, yes.

Reply:
Oh true thats a good point.
When only decimal separators occur, it is not a problem to parse both correctly!

I will change that.